public/pages/CreateTrigger/containers/CreateTrigger/utils/triggerToFormik.js

```diff
@@ -118,37 +124,46 @@ export function bucketLevelTriggerToFormik(trigger, monitor) {
   const triggerConditions = getBucketLevelTriggerConditions(condition);
   const where = getWhereExpression(composite_agg_filter);
 
+  const triggersUiMetadata = _.get(monitor, 'ui_metadata.triggers', {});
   const thresholdEnum = _.get(
-    monitor,
-    `ui_metadata.triggers[${name}].enum`,
+    triggersUiMetadata[name],
+    'enum',
     FORMIK_INITIAL_TRIGGER_VALUES.thresholdEnum
   );
   const thresholdValue = _.get(
-    monitor,
-    `ui_metadata.triggers[${name}].value`,
+    triggersUiMetadata,
+    'value',
     FORMIK_INITIAL_TRIGGER_VALUES.thresholdValue
   );
+
+  const adTriggerMetadata = _.get(triggersUiMetadata[name], 'adTriggerMetadata', {});
+  /*
+    If trigger type doesn't exist fallback to query trigger with following reasons
+    1. User has changed monitory type from normal monitor to AD monitor.
+    2. User has created / updated from API and visiting OpenSearch Dashboards to do other operations.
+   */
+  const triggerType = _.get(adTriggerMetadata, 'triggerType', TRIGGER_TYPE.ALERT_TRIGGER);
```

Comment:
This part seems to be same as above, do we need it?

Reply:
Sorry, which lines are you referring to?

Reply:
Realised they are in 2 functions, so that's fine.